pkg/tsdb/azuremonitor/httpclient.go

```diff
@@ -8,34 +8,38 @@ import (
 	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/aztokenprovider"
 )
 
-func httpClientProvider(route azRoute, model datasourceInfo, cfg *setting.Cfg) (*httpclient.Provider, error) {
-	var clientProvider *httpclient.Provider
+func getMiddlewares(route azRoute, model datasourceInfo, cfg *setting.Cfg) ([]httpclient.Middleware, error) {
+	middlewares := []httpclient.Middleware{}
 
 	if len(route.Scopes) > 0 {
 		tokenProvider, err := aztokenprovider.NewAzureAccessTokenProvider(cfg, model.Credentials)
 		if err != nil {
 			return nil, err
 		}
+		middlewares = append(middlewares, aztokenprovider.AuthMiddleware(tokenProvider, route.Scopes))
+	}
 
-		clientProvider = httpclient.NewProvider(httpclient.ProviderOptions{
-			Middlewares: []httpclient.Middleware{
-				aztokenprovider.AuthMiddleware(tokenProvider, route.Scopes),
-			},
+	if _, ok := model.DecryptedSecureJSONData["appInsightsApiKey"]; ok {
```

Comment:
Couldn't this be set together with other configured Azure services? Is there a risk `X-API-Key` interfere with requests to other services?

Reply:
That would be up to you. Just something that did catch my interest